src/rpdk/core/contract/suite/handler_create.py

```diff
@@ -54,12 +54,15 @@ def contract_invalid_create(resource_client):
 
 @failed_event(error_code=HandlerErrorCode.InvalidRequest)
 def _create_with_invalid_model(resource_client):
-    requested_model = resource_client.generate_invalid_create_example()
-    _status, response, _error_code = resource_client.call_and_assert(
-        Action.CREATE, OperationStatus.FAILED, requested_model
-    )
-    assert response["message"]
-    return _error_code
+    try:
+        requested_model = resource_client.generate_invalid_create_example()
+        _status, response, _error_code = resource_client.call_and_assert(
+            Action.CREATE, OperationStatus.FAILED, requested_model
+        )
+        assert response["message"]
+        return _error_code
+    finally:
+        resource_client.call(Action.DELETE, requested_model)
```

Comment:
```python
resource_client.call_and_assert(Action.DELETE, OperationStatus.SUCCESS, model)
```

Reply:
this is on purpose. Not all resources will create partial resource. If, the resource is not created we don't want to fail the test.

Reply:
I agree with @anshikg we should actually correct the test fixture.

Reply:
I have general thoughts around this and test error vs failure. I will create an issue with my thoughts.

Reply:
> this is on purpose. Not all resources will create partial resource. If, the resource is not created we don't want to fail the test.

makes sense, I forgot that the testing framework does not skip `NotFoundException`